### PR TITLE
[WIP] Feature/auto horizon error

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: extension/src/popup/locales/en/Please note the following the message.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (aristidesstaffieri) [inline extension/src/popup/locales/en/Please note the following the message.json:1]: Translations?

File: extension/src/background/helpers/account.ts [hunk 2]
```diff
@@ -164,6 +164,22 @@ export const getIsRpcHealthy = async (networkDetails: NetworkDetails) => {
...
 
+export const getUserNotification = async () => {
+  let response = { enabled: false, message: "" };
+
+  try {
+    const res = await fetch(`${INDEXER_URL}/user-notification`);
+    response = await res.json();
+  } catch (e) {
+    captureException(
+      `Failed to load feature flag for Soroban mainnet - ${JSON.stringify(e)}`,
```
Comment (aristidesstaffieri) [inline extension/src/background/helpers/account.ts:175]: Nit but feature flags are not per network right now

**Review by aristidesstaffieri — APPROVED** [review]
lgtm

File: extension/src/background/helpers/account.ts (hunk 2 above)
Comment (piyalbasu) [reply extension/src/background/helpers/account.ts:175]: Oh good catch - bad copy/paste job

File: @shared/api/internal.ts [hunk 3]
```diff
@@ -455,7 +455,6 @@ export const getAccountIndexerBalances = async (
   if ("error" in data && (data?.error?.horizon || data?.error?.soroban)) {
     const _err = JSON.stringify(data.error);
     captureException(`Failed to fetch account balances - ${_err}`);
-    throw new Error(_err);
```
Comment (piyalbasu) [inline @shared/api/internal.ts]: throw an error here leads to the redux thunk rejecting - which just leads to the generic `Failed to fetch account balances error`. If we don't throw, we can show the user a Horizon specific error

File: extension/src/popup/views/Account/index.tsx [hunk 4]
```diff
@@ -309,7 +309,7 @@ export const Account = () => {
                   />
                 </div>
               )}
-              {!isFunded && !hasError && (
+              {!isFunded && !hasError && !error?.horizon && (
```
Comment (piyalbasu) [inline extension/src/popup/views/Account/index.tsx:312]: don't show the unfunded message if we have a horizon message bc we're not sure if you're funded or not

**Pull request merged**